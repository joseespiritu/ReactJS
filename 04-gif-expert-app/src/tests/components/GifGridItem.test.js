import React from "react";
import GifGridItem from "../../components/GifGridItem";
import { shallow } from "enzyme";

describe("Debe de renderizar el componente correctamente", () => {
  const title = "Un titulo";
  const url = "http://localhost";
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test("Debe de ser un componente de tipo <GifGridItem/>", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de tener un pÃ¡rrafo con el title", () => {
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(title);
  });

  test("debe de tener la imagen igual al url y alt de los props", () => {
    const img = wrapper.find("img");
    // console.log(img.prop('src'));
    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
  });

  test("debe de tener animate__fadeIn", () => {
    const div = wrapper.find("div");
    const className = div.prop("className");
    expect(className.includes("animate__fadeIn")).toBe(true);
  });
});
