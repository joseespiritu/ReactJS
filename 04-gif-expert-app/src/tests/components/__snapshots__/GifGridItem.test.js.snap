// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Debe de renderizar el componente correctamente Debe de ser un componente de tipo <GifGridItem/> 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un titulo"
    src="http://localhost"
  />
  <p>
    Un titulo
  </p>
</div>
`;
